pyface/ui/qt4/code_editor/replace_widget.py

```diff
@@ -36,6 +36,11 @@ def __init__(self, parent):
 
         self.setLayout(layout)
 
+    def disconnect_event_listeners(self):
+        close_button = self.findChild(QtGui.QPushButton)
+        if close_button is not None and close_button.text() == "Close":
+            close_button.clicked.disconnect(self.hide)
+
```

Comment:
I thought we were going to store the reference to the button and then get rid of it like in the other PRs. 
If we can't do this confidently, let's not do it then.
I am sorry I assumed we resolved this in the other PRs and that could be extended here. I should have commented about it again while this was a draft.

Reply:
I don't think `_connections_to_remove` would work here because `ReplaceWidget.__init__()` calls `FindWidget.__init__()` which sets up a button in the layout that is then overwritten by the `ReplaceWidget` layout. Unless we save only one `connection_to_remove` that is also overwritten in `ReplaceWidget.__init__()`.

Reply:
OK, that looks complicated. Let's not do that then.

Reply:
Went with the 3rd approach and removed `close_button` disconnects completely